### PR TITLE
refactor: move bin prefix logic to new file

### PR DESCRIPTION
**Pull request by istae**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2468)

### CODE REVIEW

File: pkg/topology/kademlia/binprefix.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package kademlia
```
Comment (acud) [inline pkg/topology/kademlia/binprefix.go:1]: copyright missing. also, if there are unit tests for this code (hopefully there are specific ones) we could move them to a separate test file too

Comment (istae) [reply pkg/topology/kademlia/binprefix.go:1]: surprisingly, there are no tests that check that the correct prefixes are generated.

**Review by mrekucci — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2468)* status: all files reviewed, 1 unresolved discussion (waiting on @acud)

File: pkg/topology/kademlia/binprefix.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"math"
+	"math/bits"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// peerConnInfo groups necessary fields needed to create a connection.
```
Comment (acud) [inline pkg/topology/kademlia/binprefix.go:14]: wrong comment

**Review by acud — APPROVED** [review]

**Pull request merged**